lib/enrollment_repository.rb

```diff
@@ -1,52 +1,48 @@
 require "csv"
-require "pry"
 require_relative "enrollment"
 require_relative "district"
-require 'json'
-require 'ostruct'
 
 class EnrollmentRepository
-  attr_accessor :enrollments
-
   def initialize
     @enrollments = {}
   end
 
-  def hash_pathway
-    hash_pathway = {"kindergarten" => "kindergarten_participation",
-                  "high_school_graduation" => "high_school_graduation_rates" }
-  end
-
-  def load_data(data)
-    hash_pathway.each do |key, percentages|
+  def load_enrollment_data(array_of_attr_path_hashes)
+    array_of_attr_path_hashes.each do |key_path_hash|
       collection = Hash.new
+      csv_file = key_path_hash[key_path_hash.keys[0]]
 
-      CSV.foreach(data[:enrollment][key.to_sym],
-                       headers: true, header_converters: :symbol) do |row|
-        collection[row[:location]] ||= Hash.new
-        collection[row[:location]][row[:timeframe].to_i] = row[:data][0..4].to_f
-      end
+      CSV.foreach(csv_file, headers: true, header_converters: :symbol) do |row|
+        district_name = row[:location]
+        timeframe = row[:timeframe].to_i
+        data = row[:data]
+
+        collection[district_name] ||= Hash.new
+        collection[district_name][timeframe] = data[0..4].to_f
+      end # CSV.foreach
 
-      collection.each do |location, data|
-            enrollment = Enrollment.new(name: location,
-                                           percentages.to_sym => data)
-        if enrollments[location]
-            enrollments[location].school_data[percentages.to_sym] = data
+      collection.each do |district_name, data|
+        enrollment = Enrollment.new(name: district_name,
+                                    key_path_hash.keys[0] => data)
+        if @enrollments[district_name]
+          enrollment_data_key = "#{key_path_hash.keys[0].to_s}="
+          @enrollments[district_name].send(enrollment_data_key, data)
         else
-            enrollments[location] = enrollment
-        end
-      end
-    end
-  end
+          @enrollments[district_name] = enrollment
+        end # if enrollments[district_name]
+      end # collection.each
+    end # array_of_attr_path_hashes.each
+  end # load_data
 
   def find_by_name(name)
     if name.nil?
       nil
     else
-      data = enrollments.find do |element|
-       element[1].name.downcase == name.downcase
-    end
-      data && data[1]
+      enrollment = @enrollments.find do |element|
```

Comment:
We are being clear about what kind of data we're returning here. Good variable names are key.